SeaPublicWebsite/ExternalServices/BreApi.cs

```diff
@@ -0,0 +1,95 @@
+                    string password = Global.BrePassword;
+                    Guid nonce =  Guid.NewGuid();
+                    string created = DateTime.Now.ToUniversalTime().ToString
+                        (DateTimeFormatInfo.InvariantInfo.SortableDateTimePattern) + "Z";
+                    string token = GenerateToken(password + nonce + username + created);
+                    string wsseHeader =
+                        $"WSSE UsernameToken Token=\"{token}\", Nonce=\"{nonce}\", Username=\"{username}\", Created=\"{created}\"";
+                    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", wsseHeader);
+
+                    string path = "/bemapi/energy_use";
+                    string requestString = JsonConvert.SerializeObject(request);
+                    StringContent stringContent = new(requestString);
+                    HttpResponseMessage response = await Task.FromResult(httpClient.PostAsync(path, stringContent)).Result;
+
+                    if (response.IsSuccessStatusCode)
+                    {
+                        string bodyString = await Task.FromResult(response.Content.ReadAsStringAsync().Result);
+                        JObject measures = JObject.FromObject(JObject.Parse(bodyString)["measures"] ?? new JObject());
+
+                        List<Recommendation> recommendations = new List<Recommendation>();
+                        foreach (JProperty prop in measures.Properties())
+                        {
+                            Dictionary<string, Recommendation> recommendationDictionary =
+                                RecommendationService.RecommendationDictionary;
+                            JToken value = prop.Value;
+                            int minInstallCost = (int) value["min_installation_cost"];
+                            int maxInstallCost = (int) value["max_installation_cost"];
+                            int saving = (int) value["cost_saving"];
+                            int lifetime = (int) value["lifetime"];
+
+                            Recommendation recommendation = new()
+                            {
+                                Key = recommendationDictionary[prop.Name].Key,
+                                Title = recommendationDictionary[prop.Name].Title,
+                                MinInstallCost = minInstallCost,
+                                MaxInstallCost = maxInstallCost,
+                                Saving = saving,
+                                LifetimeSaving = lifetime * saving,
+                                Lifetime = lifetime,
+                                Summary = recommendationDictionary[prop.Name].Summary
+                            };
+                            recommendations.Add(recommendation);
+                        }
+                        return recommendations;
+                    }
+
+                    return null;
```

Comment:
This null looks like it will blow up into a null ref exception further up the stack. We should handle BRE API failures more explicitly. Potentially by throwing an exception here so the user sees a more helpful error message (we don't have a nice error page yet though).
Same for catching an exception below.